src/config/config.cc

```diff
@@ -11,16 +11,29 @@
 
 namespace cris::core {
 
+// Using macro for defining multiple functions.
+// `type` is a typename so that it cannot be in parentheses.
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage,-warnings-as-error)
 #define __CRIS_CORE_DEFINE_CONFIG_TYPE(type)                                                  \
+    /* `type` is a typename so that it cannot be in parentheses.      */                      \
```

Comment:
Can we alias type so that it becomes usable?
One possibility is to have all these in an anonymous ns first, and `using` right afterward if needed to take interface out, so that you can safely alias whatever type with in that ns.

Reply:
I don't understand what do you mean.

Here the problem is, tidy checks require parentheses around `type`, how do you aliasing to workaround? Using a specific type as an example may help me understand better.

Reply:
```
using type_alias = type;

// From now on only use type_alias, which is not a macro arg.
```

Reply:
I won't worry about leaking alias out as long as it's well-hidden, and it provides a lot more benefit of simplifying the code than the issue it can have.
For example, you can even write "you_are_not_supposed_to_name_anything_like_this_manually" in var name, so that it's almost idiot-proof.

Anyway, up to you, just some suggestions.

Reply:
You probably need different alias for different `type`, so you will need string concatenation when defining alias. However, `type` is not necessary to be just letters and numbers, it may contains ':' '<' '>' or so, so you need to handle those edge cases, and that sounds like an overkill to me.